open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/src/main/java/org/odpi/openmetadata/accessservices/assetlineage/handlers/AssetContextHandler.java

```diff
@@ -134,20 +134,19 @@ private void buildAssetContext(String userId, EntityDetail entityDetail,String s
             addContextForSchemaAttributeType(userId,entityDetail,typeDefName);
         }
 
-        if(!isComplexSchemaType(userId,typeDefName)) {
             List<EntityDetail> tableTypeEntities = buildGraphByRelationshipType(userId, entityDetail, ATTRIBUTE_FOR_SCHEMA, typeDefName, false);
 
+            if(tableTypeEntities.isEmpty()){
+              tableTypeEntities = buildGraphByRelationshipType(userId, entityDetail, NESTED_SCHEMA_ATTRIBUTE, typeDefName, false);
+            }
             for (EntityDetail schemaTypeEntity : tableTypeEntities) {
                 if (isComplexSchemaType(userId, schemaTypeEntity.getType().getTypeDefName())) {
                     setAssetDetails(userId, schemaTypeEntity);
                 } else {
-                    List<EntityDetail> tableEntities = buildGraphByRelationshipType(userId, schemaTypeEntity, SCHEMA_ATTRIBUTE_TYPE, typeDefName, false);
-                    if (!CollectionUtils.isEmpty(tableEntities)) {
-                        buildAssetContext(userId, tableEntities.stream().findFirst().get(), superType);
-                    }
+                        buildAssetContext(userId, tableTypeEntities.stream().findFirst().get(), superType);
```

Comment:
get() might return NoSuchElementException if the Optional is empty. 
Also, it doesn't look like the superType parameter is used anywhere in the method